test-data/unit/check-isinstance.test

```diff
@@ -1425,4 +1434,17 @@ T = TypeVar('T', bound=A)
 def f(x: T) -> None:
     if isinstance(x, B):
         reveal_type(x) # E: Revealed type is '__main__.B'
+    else:
+        reveal_type(x) # E: Revealed type is 'T`-1'
+    reveal_type(x) # E: Revealed type is 'T`-1'
 [builtins fixtures/isinstance.pyi]
+
+[case testIsinstanceAndTypeType]
+from typing import Type
+def f(x: Type[int]) -> None:
+    if isinstance(x, type):
+        reveal_type(x) # E: Revealed type is 'Type[builtins.int]'
+    else:
+        reveal_type(x)  # Unreachable
```

Comment:
That could be nice. There are some cases when this is valid, foe example when type checking `AnyStr` some blocks are often unreachable for either `str` or `bytes`, so the required logic may be somewhat involved.